### DISCUSSION TITLE
Using Homebrew for MW4 Installation

### DISCUSSION THREAD

**Discussion by Altair8861** [mworion/MountWizzard4#252] [Q&A] [answered]
Having a very difficult time getting things set up on my iMac.

Wondering if this can be done with Homebrew?

**Comment by mworion** [comment, marked as answer]
Hi, 
downloading python from python.org should work on your Mac.
If you have a Mac with M1 / M2 CPU, you need to use Rosetta emulator, as the python support with all packages direct on M1 silicon is not there yet.
https://www.astroworldcreations.com/blog/apple-silicon-and-legacy-command-line-software
Hope this helps
Michel

**Reply by Altair8861** [reply]
Thanks Michel. Performed change on Terminal Prefs as per the web page. Python 3.9.13 installed, verified by --version command. This procedure takes my Terminal skills to the limit ;-)

I copied the three files in the support folder to a work folder as per the Youtube video. Double clicking on the MW4_Install.command.html script does not run an install process, it just takes me back to the github page.

**Comment by mworion** [comment]
Hi,
Sorry to hear.
I guess that you downloaded only the link to the scripts, not the scripts themselves. Ideally you download the ZIP file and extract all files locally (probably you did it). If you could post screenshots of the terminal window this could help. Do you own a M1 silicon Mac? 
Michel

**Reply by Altair8861** [reply]
I made another attempt today following the YouTube video "mw4_install_macos" step by step.

From GitHub I downloaded Support->2.0->MacOSx_scripts.zip

The zip file contents were extracted and placed in the folder Mistress->Applications->MW4files

I opened the MW4_Install.command and it seemed to work - I saw a venv folder created.

I opened the MW4_run.command file and I did not get an application launch like in the Youtube video. Rather, another Terminal window popped up.

Checking the install.log file there appears to be a problem:

× Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      Traceback (most recent call last):
        File "/Applications/MW4Files/venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/Applications/MW4Files/venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Applications/MW4Files/venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/private/var/folders/r2/01vdjrn920lfgsgxp5_1_mtw0000gn/T/pip-build-env-2o80f7yn/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 177, in get_requires_for_build_wheel
          return self._get_build_requires(
        File "/private/var/folders/r2/01vdjrn920lfgsgxp5_1_mtw0000gn/T/pip-build-env-2o80f7yn/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 159, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/r2/01vdjrn920lfgsgxp5_1_mtw0000gn/T/pip-build-env-2o80f7yn/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 174, in run_setup


Attached are screen shots after opening each .command file, plus the install.log file

![After Install Command](https://user-images.githubusercontent.com/110138869/182259699-0f1e31f4-d2b4-463d-b689-8d35c3a79a10.jpg)
![After Run Command](https://user-images.githubusercontent.com/110138869/182259708-a61a13ae-212d-4c95-8a54-be7299b162b0.jpg)
[install.log](https://github.com/mworion/MountWizzard4/files/9237849/install.log)

**Reply by Altair8861** [reply]
Forgot to mention - the links on this page:

https://mountwizzard4.readthedocs.io/en/latest/

Lead to this page:

https://github.com/mworion/MountWizzard4/blob/master/support/MacOSx_Scripts.zip

So I am not entirely sure I downloaded the correct files.

**Reply by mworion** [reply]
I think you have the right files. Do you use Mac with M1 silicon ? If so, you have to start the command files inside a terminal with Rosetta emulation as described above. 
If you don't use Apples silicon, please post the complete log files by dropping them here.
Michel

**Reply by Altair8861** [reply]
Yes, I am using the M1 IMac.

I run the command files from Terminal, I was double-clicking them in the Finder.

**Comment by Altair8861** [comment]
Just ran the files from inside the Terminal window, using the Rosetta emulations - no luck. Screen shots, install log, and run log attached.
![Install Command Terminal Window](https://user-images.githubusercontent.com/110138869/182500636-718f1623-c72d-4a88-8196-82498bf203c0.jpg)
![Run Command Terminal Window](https://user-images.githubusercontent.com/110138869/182500640-0d39180d-28c4-4bd8-9e14-21b8549c35e4.jpg)
[install.log](https://github.com/mworion/MountWizzard4/files/9247027/install.log)
[run.log](https://github.com/mworion/MountWizzard4/files/9247028/run.log)

**Reply by mworion** [reply]
...I tried to remove some obstacles from script. Please unzip and try Install from terminal again
[MaxOSx_Scripts.zip](https://github.com/mworion/MountWizzard4/files/9252766/MaxOSx_Scripts.zip)
Michel

**Reply by Altair8861** [reply]
Late last night I successfully resolved the problem.

1) Deleted Python 3.10.x
2) Installed Python 3.9.x
3) Ran scripts from inside Terminal.

I suspect my first Python install was not correct or complete.

It worked just like in your video. Got it running on my Linux machine too.

To be safe, I will download the new scripts, thank you!

**Comment by mworion** [comment]
Jeff, great to hear, have fun !
Michel
